src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -267,15 +266,6 @@ class PeerActor(
       messageSubscribers = messageSubscribers.filterNot(_.ref == sender())
   }
 
-  def handlePeerChainCheck(rlpxConnection: RLPxConnection): Receive = {
```

Comment:
I think you are right, `FastSyncHost` could also be used for this request :+1:. However, as in our client we first ask for the fork block before fully connecting to a peer, shouldn't we able to respond to the same request for the fork block during this setup phase? For example, before the removal of this function (`handlePeerChainCheck`) we also used it in `waitingForForkHeader`, but we currently only handle the block requests (with `FastSyncHost`) in the `MessageHandler`.
Maybe we should also rename `FastSyncHost` if it's used outside of FastSync